tests/js/client/active-failover/basic.js

```diff
@@ -246,6 +247,36 @@ function checkForFailover(leader) {
   throw "No failover occured";
 }
 
+function waitUntilHealthStatusIs(isHealthy, isFailed) {
+  print("Waiting for health status to be healthy: ", JSON.stringify(isHealthy), " failed: ", JSON.stringify(isFailed));
+  // Wait 25 seconds, sleep 5 each run
+  for (const start = Date.now(); (Date.now() - start) / 1000 > 25; internal.wait(5.0)) {
+    let needToWait = false;
+    let res = readAgencyValue("/arango/Supervision/Health");
+    let srvHealth = res[0].arango.Supervision.Health;
+    for (const [key, srv] of Object.entries(srvHealth)) {
```

Comment:
This function doesn't quite do what I'd expect. It checks for all entries in the agency whether they actually exist in the right list (isFailed or isHealthy). I'd have expected it to check for all entries in `isFailed` and `isHealthy` that they're in `Supervision/Health` with the right state.

This makes a difference whenever the list of servers in the agency isn't equal to the union of `isHealthy` and `isFailed`.

Reply:
Right, I need to adjust!
Good spot thanks